### PR TITLE
Proposal: Binlog replication(syncer) unit support plugin

### PR DESCRIPTION
**Pull request by WangXiangUSTC**
### What problem does this PR solve? 

Add a proposal for supporting plugin in binlog replication(syncer) unit

### CODE REVIEW

File: docs/RFCS/20200412_syncer_plugin.md [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+- When the dml job execute success, then send it to other platform.
+
+### Hook
+
+`Init` can be execute when [initial](https://github.com/pingcap/dm/blob/9023c789964fde0f5134e0c49435db557e21fdf7/syncer/syncer.go#L257) binlog replication unit.
+
+`HandleDDLJobResult` handle the result of [handleQueryEvent](https://github.com/pingcap/dm/blob/9023c789964fde0f5134e0c49435db557e21fdf7/syncer/syncer.go#L1279) and then do something.
+
+`HandleDMLJobResult` handle the result of [handleRowsEvent](https://github.com/pingcap/dm/blob/9023c789964fde0f5134e0c49435db557e21fdf7/syncer/syncer.go#L1274) and then do something.
+``
```
Comment (csuzhangxc) [inline docs/RFCS/20200412_syncer_plugin.md:76]: redundancy ``` `` ```.

File: docs/RFCS/20200412_syncer_plugin.md [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+
+This interface used to handle the DML job's result in binlog replication unit(syncer)
+
+- When the dml job execute success, then send it to other platform.
+
+### Hook
+
+`Init` can be execute when [initial](https://github.com/pingcap/dm/blob/9023c789964fde0f5134e0c49435db557e21fdf7/syncer/syncer.go#L257) binlog replication unit.
+
+`HandleDDLJobResult` handle the result of [handleQueryEvent](https://github.com/pingcap/dm/blob/9023c789964fde0f5134e0c49435db557e21fdf7/syncer/syncer.go#L1279) and then do something.
```
Comment (csuzhangxc) [inline docs/RFCS/20200412_syncer_plugin.md:73]: NOTE: we use `runFatalChan` to report errors between goroutines.

File: docs/RFCS/20200412_syncer_plugin.md (hunk 1 above)
Comment (WangXiangUSTC) [reply docs/RFCS/20200412_syncer_plugin.md]: removed

File: docs/RFCS/20200412_syncer_plugin.md (hunk 2 above)
Comment (WangXiangUSTC) [reply docs/RFCS/20200412_syncer_plugin.md:73]: I add Note in 3da265a

File: docs/RFCS/20200412_syncer_plugin.md [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+
+#### HandleDMLJobResult
+
+This interface used to handle the DML job's result in binlog replication unit(syncer)
+
+- When the dml job execute success, then send it to other platform.
+
+### Hook
+
+`Init` can be execute when [initial](https://github.com/pingcap/dm/blob/9023c789964fde0f5134e0c49435db557e21fdf7/syncer/syncer.go#L257) binlog replication unit.
```
Comment (csuzhangxc) [inline docs/RFCS/20200412_syncer_plugin.md:71]: Is it better to use a relative path for the repo? ditto for L73 and L75.

Comment (WangXiangUSTC) [reply docs/RFCS/20200412_syncer_plugin.md:71]: but the code may be changed, and the line will be wrong

Comment (csuzhangxc) [reply docs/RFCS/20200412_syncer_plugin.md:71]: oh...

File: docs/RFCS/20200412_syncer_plugin.md [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+
+`HandleDDLJobResult` handle the result of [handleQueryEvent](https://github.com/pingcap/dm/blob/9023c789964fde0f5134e0c49435db557e21fdf7/syncer/syncer.go#L1279) and then do something.
+
+`HandleDMLJobResult` handle the result of [handleRowsEvent](https://github.com/pingcap/dm/blob/9023c789964fde0f5134e0c49435db557e21fdf7/syncer/syncer.go#L1274) and then do something.
+
+NOTE: We use `runFatalChan` to report errors between goroutines for executing ddl and dml, and the `err` of `handleQueryEvent` and `handleRowsEvent` don't contain the real error messag. Need to refine it.
```
Comment (lichunzhu) [inline docs/RFCS/20200412_syncer_plugin.md:77]:
```suggestion
NOTE: We use `runFatalChan` to report errors between goroutines for executing ddl and dml, and the `err` of `handleQueryEvent` and `handleRowsEvent` don't contain the real error message. Need to refine it.
```

Comment (lichunzhu) [inline docs/RFCS/20200412_syncer_plugin.md:77]: https://github.com/pingcap/dm/pull/605. The atomic error `execError` used in this PR may fix this problem 🤔

Comment (WangXiangUSTC) [reply docs/RFCS/20200412_syncer_plugin.md]: ok, I delete this note in 4e6b864

**Pull request merged**